tests/sync/test_syncing.py

```diff
@@ -91,3 +136,151 @@ def test_complex_sync_all(self, tmpdir: py.path.local):
             pathlib.Path(f'{tmpdir}/syncs/sync-2/group4-file1.txt'),
             pathlib.Path(f'{tmpdir}/syncs/sync-2/group4-file2.txt'),
         ]
+
+
+class TestConfigError:
+    def test_valid_configuration(self, tmpdir: py.path.local):
+        file_name = pathlib.Path(tmpdir / 'config.yml')
+        with file_name.open('w') as f:
```

Comment:
`py.path.local` is already something like `pathlib.Path` (but before that existed...). You should just be able to use `config_yml = tmpdir / 'config.yml` and then `config_yml.write_text(..., encoding='utf-8')` instead ([docs](http://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.write_text))

(same thing in other places too)

Reply:
I also would propose using `config_yml` or so instead of `file_name` because it's not really a filename.

Reply:
That makes the code cleaner, thanks.

I had to keep it in the test where the file does not exist, because it raises a different error if it's not a `Path` object